src/utils/utils.js

```diff
@@ -80,4 +80,8 @@ export const warnAboutDepreation = (deprecatedParam, useInstead) => {
  */
 export const callIfFunction = (arg) => typeof arg === 'function' ? arg() : arg
 
+export const hasToPromiseFn = (arg) => arg && typeof arg.toPromise === 'function'
+
 export const isPromise = (arg) => arg && Promise.resolve(arg) === arg
+
+export const promisify = (arg) => hasToPromiseFn(arg) ? arg.toPromise() : Promise.resolve(arg)
```

Comment:
The verb "promisify" has been used a lot to describe converting a callback-based API to a Promise-based API. For example, NodeJS's [util.promisify](https://nodejs.org/dist/latest-v8.x/docs/api/util.html#util_util_promisify_original), Bluebird's [Promise.promisify](http://bluebirdjs.com/docs/api/promise.promisify.html), [pify](https://www.npmjs.com/package/pify), and more. I suggest renaming to something else. Ideas: `toPromise` or `asPromise`.

Reply:
I've updated it to `asPromise`.